package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -57,6 +57,9 @@ def get_dataset(self, dataset_name: str) -> Optional[AbstractDataSet]:
                 dataset_obj = self._catalog._get_dataset(dataset_name)
         except DataSetNotFoundError:  # pragma: no cover
             dataset_obj = None
+        # if dataset has no catalog entry, it is a MemoryDataSet
+        if not dataset_obj:
+            dataset_obj = MemoryDataSet()
```

Comment:
Is this any different from the below?
```suggestion
            dataset_obj = MemoryDataSet()
```

Reply:
it work's if you just add it in the exception. the reason why it flickered was because filepath is none for MemoryDataSets and it is a required field in DataNodeResponseAPI which now I have made Optional. 

I will make the above change :)

Reply:
I'm happy with this. Yeah, I just didn't understand/get to finding what was happening with the validation error, but this solves it.